node/lib/util/push.js

```diff
@@ -93,24 +94,23 @@ exports.push = co.wrap(function *(repo, remoteName, source, target, force) {
     // First, push the submodules.
 
     let errorMessage = "";
-    const subRepos = yield SubmoduleUtil.getSubmoduleRepos(repo);
     const shas = yield SubmoduleUtil.getSubmoduleShasForBranch(repo, source);
-    const index = yield repo.index();
-    const urls = yield SubmoduleConfigUtil.getSubmodulesFromIndex(repo, index);
```

Comment:
I don't think I agree -- which configured remote should we use? I think it's reasonable to allow an option, as we discussed, or other configuration to help guide which remote(s) to fetch from; this will become especially relevant if we opt to go with "true" forking at some point.

Either way, I think that discussion is orthogonal to whether or not to grab the URL from the index or the commit.